crates/core/src/diff/fragment/tests.rs

```diff
@@ -1,7 +1,226 @@
+  <Column width="fill" verticalArrangement="center" horizontalAlignment="center" scroll="vertical">
+    <OutlinedButton phx-click="showDialog"><Text>Show Dialog</Text></OutlinedButton>
+
+    <AlertDialog phx-click="hideDialog">
+      <ConfirmButton>
+          <TextButton phx-click="hideDialog">
+            <Text>Confirm</Text>
+          </TextButton>
+      </ConfirmButton>
+      <DismissButton>
+        <OutlinedButton phx-click="hideDialog">
+          <Text>Dismiss</Text>
+        </OutlinedButton>
+      </DismissButton>
+      <Icon imageVector="filled:Add"></Icon>
+      <Title>Alert Title</Title>
+      <Content>
+          <Text>Alert message</Text>
+      </Content>
+    </AlertDialog>
+
+    <Box size="100" contentAlignment="center">
+      <BadgeBox containerColor="\#FF0000FF" contentColor="\#FFFF0000">
+        <Badge><Text>+99</Text></Badge>
+        <Icon imageVector="filled:Add"></Icon>
+      </BadgeBox>
+    </Box>
+    <ElevatedButton phx-click="showDialog"><Text>ElevatedButton</Text></ElevatedButton>
+    <FilledTonalButton phx-click="showDialog"><Text>FilledTonalButton</Text></FilledTonalButton>
+    <TextButton phx-click="showDialog"><Text>TextButton</Text></TextButton>
+
+  </Column>
+</Scaffold>"#;
+    assert_eq!(out, expected);
+    let new_document = Document::parse(out).expect("Failed to parse rendered dom");
+    let patches = crate::diff::diff(&document, &new_document);
+    if patches.is_empty() {
+        return;
+    }
+
+    let mut editor = document.edit();
+    let mut stack = vec![];
+    for patch in patches.into_iter() {
+        let _ = patch.apply(&mut editor, &mut stack);
+    }
+    editor.finish();
+    //document.merge_fragment(diff.clone()).expect("Failed to merge in diff with document");
+    let document_expected = r#"<Scaffold>
+    <TopAppBar>
+        <Title>
+            <Text>
+                Hello
+            </Text>
+        </Title>
+    </TopAppBar>
+    <FloatingActionButton phx-click="inc">
+        <Icon imageVector="filled:Add" />
+    </FloatingActionButton>
+    <Column width="fill" verticalArrangement="center" horizontalAlignment="center" scroll="vertical">
+        <OutlinedButton phx-click="showDialog">
+            <Text>
+                Show Dialog
+            </Text>
+        </OutlinedButton>
+
+        <AlertDialog phx-click="hideDialog">
+            <ConfirmButton>
+                <TextButton phx-click="hideDialog">
+                  <Text>Confirm</Text>
+                </TextButton>
+            </ConfirmButton>
+            <DismissButton>
+              <OutlinedButton phx-click="hideDialog">
+                <Text>
+                    Dismiss
+                </Text>
+              </OutlinedButton>
+            </DismissButton>
+            <Icon imageVector="filled:Add"></Icon>
+            <Title>Alert Title</Title>
+            <Content>
+                <Text>Alert message</Text>
+            </Content>
+        </AlertDialog>
```

Comment:
The main bug here is that this `AlertDialog` is 
```
 <AlertDialog phx-click="hideDialog">
    <ConfirmButton>
        <TextButton phx-click="hideDialog">
            <Text>
                Confirm
            </Text>
        </TextButton>
   </ConfirmButton>
</AlertDialog>
```

when it should be what's shown in this test case.